agents/holo_fpga/roach_agent.py

```diff
@@ -0,0 +1,182 @@
+        self.lock = TimeoutLock()
+
+        """
+        ### register the position feeds
+
+        ### Agent registers data feed, things published to feed go to grafana and .g3 files
+        ### pick a good name for feeds, once it's registered it's kinda permanent
+
+        """
+        agg_params = {"frame_length": 10 * 60}  # [sec]
+        self.agent.register_feed(
+            "fpga", record=True, agg_params=agg_params, buffer_time=0
+        )
+
+        # Load dictionary of specific mirror paramters, since some parameters
+        # like limits and translate vary over different FTSes This is loaded
+        # from a yaml file, which is assumed to be in the $OCS_CONFIG_DIR
+        # directory.
+        if config_file is None:
+            raise Exception("No config file specified for the FTS mirror config")
+        else:
+            config_file_path = os.path.join(os.environ["OCS_CONFIG_DIR"], config_file)
+            with open(config_file_path) as stream:
+                self.holog_configs = yaml.safe_load(stream)
+                if self.holog_configs is None:
+                    raise Exception("No mirror configs in config file.")
+                self.log.info(f"Loaded mirror configs from file {config_file_path}")
+                self.baseline = self.holog_configs.pop("baseline", None)
+                self.roach = self.holog_configs.pop("roach", None)
+
+                # The other mirror configs (speed, timeout) are optional and
+                # have defaults so we leave them as the dictionary.
+                if self.roach is None or self.baseline is None:
+                    raise Exception("IP address and channels must be included "
+                                    "in the holography configuration keys.")
+
+        self.roach, self.opts, self.baseline = fpga_daq3.roach2_init()
+
+        err = os.system("/opt/anaconda2/bin/python2 ~/Desktop/holog_daq/scripts/upload_fpga_py2.py")
+        assert err == 0
+
+        print("Connecting to server %s ... " % (self.roach))
+
+        self.fpga = casperfpga.CasperFpga(self.roach)
+        time.sleep(1)
+
+        if self.fpga.is_connected():
+            print("ok\n")
+
+    def take_data(self, session, params=None):
+        """take_data(params=None)
+
+        **Task** - A task to take data from the FPGA.
+
+        Parameters: None
+
+        Examples:
+            Example for calling in a client::
+
+                import ocs
+                from ocs.ocs_client import OCSClient
+                agent_fpga = OCSClient("fpga") # create agent
+                agent_fpga.take_data() # take data
+
+
+        Notes:
+            An example of the session data::
+
+                >>> response.session['data']
+                {"fields":
+                    {"timestamp": 1601924482.722671, "block_name": "fpga", "data":{"amp_AA": 293.644, "amp_BB": 33.752, "amp_AB": 33.752, "arr_P": 33.752}
+                    }
+                }
+        """
```

Comment:
A few things in this docstring:
- The first line should be changed from `take_data(params=None)` to `take_data()`. We want to document this from the Client's perspective, which doesn't really know about the `params` dictionary. When calling this from a Client (pretending it had an argument) a user is going to do something like `client.take_data(arg1=7)` not `client.take_data(params={'arg1': 7})`.
- Remove the `Parameters: None` line
- Remove `import ocs` from the Example code block, it's not needed
- The `session.data` example has an extra nested level that isn't there in the code, you can remove the `{"fields":` line and it's corresponding `}`. This example should exactly match the format you're passing to `session.data`. You can confirm this format a number of ways, one is to run your Agent, and use a client to run the `take_data` task, if you then run:
    ```
    client.take_data.status().session['data']
    ```
    You should see this `session.data` object. Again, that should match what's shown here.